animate/adapt.py

```diff
@@ -111,27 +138,84 @@ def interpolate(self, f):
         Interpolate a :class:`.Function` into the corresponding :class:`.FunctionSpace`
         defined on the adapted mesh.
 
-        :param: the scalar :class:`.Function` on the initial mesh
-        :return: its interpolation onto the adapted mesh
+        :arg f: a Function on the initial mesh
+        :type f: :class:`firedrake.function.Function`
+        :returns: its interpolation onto the adapted mesh
+        :rtype: :class:`firedrake.function.Function`
         """
         raise NotImplementedError(
             "Consistent interpolation has not yet been implemented in parallel"
         )  # TODO
 
 
-def adapt(mesh, *metrics, name=None):
+def adapt(mesh, *metrics, name=None, serialise=False, remove_checkpoints=True):
     r"""
     Adapt a mesh with respect to a metric and some adaptor parameters.
 
     If multiple metrics are provided, then they are intersected.
 
-    :param mesh: :class:`~firedrake.mesh.MeshGeometry` to be adapted.
-    :param metrics: list of :class:`.RiemannianMetric`\s
-    :param name: name for the adapted mesh
-    :return: a new :class:`~firedrake.mesh.MeshGeometry`.
+    :arg mesh: mesh to be adapted.
+    :type mesh: :class:`firedrake.mesh.MeshGeometry`
+    :arg metrics: metrics to guide the mesh adaptation
+    :type metrics: :class:`list` of :class:`.RiemannianMetric`\s
+    :kwarg name: name for the adapted mesh
+    :type name: :class:`str`
+    :kwarg serialise: if ``True``, adaptation is done in serial
+    :type serialise: :class:`bool`
+    :kwarg remove_checkpoints: if ``True``, checkpoint files are deleted after use
+    :type remove_checkpoints: :class:`bool`
+    :returns: the adapted mesh
+    :rtype: :class:`~firedrake.mesh.MeshGeometry`
     """
+
+    # Parallel adaptation is currently only supported in 3D
+    if mesh.topological_dimension() != 3:
+        serialise = False
+
+    # If already running in serial then no need to use checkpointing
+    if COMM_WORLD.size == 1:
+        serialise = False
+
+    # Combine metrics by intersection, if multiple are passed
     metric = metrics[0]
     if len(metrics) > 1:
         metric.intersect(*metrics[1:])
+
     adaptor = MetricBasedAdaptor(mesh, metric, name=name)
-    return adaptor.adapted_mesh
+    if serialise:
+        checkpoint_dir = get_checkpoint_dir()
+        if not os.path.exists(checkpoint_dir):
+            os.makedirs(checkpoint_dir)
+        metric_name, mesh_name = "tmp_metric", "tmp_adapted_mesh"
```

Comment:
--- Marking all connected comments with these dashes---

`mesh_name` here is the name of the mesh that the user defined in their script, so should be `mesh_name = mesh.name`

Reply:
What about the change in 61717de?

Reply:
Sorry, that doesn't work, no. The `name` that we pass to `new_mesh = adapt(old_mesh, metric, name=name)`) is the name of the adapted mesh: `new_mesh.name`. And `mesh_name` in L189 is `old_mesh.name`, which the user might have set to something when constructing the mesh, e.g. `old_mesh = RectangleMesh(..., name='original_mesh')`.

So when you do 
```
with fchk.CheckpointFile(fname, "r", comm=comm) as chk:
    mesh = chk.load_mesh()
```
in `load_checkpoint`, this assumes by default `mesh = chk.load_mesh('firedrake_default')`, whereas we need to do `mesh = chk.load_mesh('original_mesh')`

Reply:
Fixed in 7fcce996cd69fb074e6bf6657c6346631053461f . See https://github.com/pyroteus/animate/pull/72#discussion_r1517913907 for MFE before the fix.